pkg/remotecmd/interface.go

```diff
@@ -0,0 +1,51 @@
+package remotecmd
+
+import (
+	devfilev1 "github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
+
+	"github.com/redhat-developer/odo/pkg/kclient"
+)
+
+// RemoteProcessStatus is an enum type for representing process statuses.
+type RemoteProcessStatus int
+
+const (
+	// Unknown represents a process for which the status cannot be determined reliably or is not handled yet by us.
+	Unknown RemoteProcessStatus = iota + 1
```

Comment:
I was referring to this in the Coding Conventions page : https://github.com/redhat-developer/odo/wiki/Dev:-Coding-Conventions#start-enums-at-one
Because the zero value does have much value here.

Reply:
I was referring to "Learning Go" by Jon Bodner and here's an excerpt from it that made me ask in the first place:
> If there isn't a sensical default value for your constants, a common pattern is to assign the first `iota` value in the constant block to a `_` or to a constant that indicates the value is invalid.

The `Unknown` we have defined here seemed a lot closer to what Jon refers to as "invalid" in the above statement. Anyway, what you pointed to makes sense as well. There's no right or wrong here, IMO. I was only curious, and hence the question. :)

Reply:
Whoops - for some reason, my reply did not get sent. Putting it below for the record :-)

> Interesting - thanks for sharing this. "Unknown" is actually a valid state to me, which we might want to know about. For now, it is coupled with an `error` returned. But while I prefer the `iota + 1` approach (because it reads easily IMO), both recommendations look fine, because the result (starting Enums at one) is the same in the end. I'll add this point to the Coding Conventions wiki page.